File: android/src/main/java/com/cactuslab/capacitor/nativenavigation/types/ComponentOptions.java
package com.cactuslab.capacitor.nativenavigation.types;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.List;

public class ComponentOptions {
    @Nullable
    public String title;
    @Nullable
    public StackOptions stack;
    @Nullable
    public TabOptions tab;

    @Nullable
    public ModalPresentationStyle modalPresentationStyle;

    public static class StackOptions {
        @Nullable
        public StackItem backItem;
        @Nullable
        public List<StackItem> leftItems;
        @Nullable
        public List<StackItem> rightItems;
    }

    public static class StackItem {
        @NonNull
        public String id;
        @NonNull
        public String title;
        @Nullable
        public String image;

        public StackItem(@NonNull String id, @NonNull String title, @Nullable String image) {
            this.id = id;
            this.title = title;
            this.image = image;
        }
    }

    public static class TabOptions {
        @Nullable
        public String image;
        @Nullable
        public String badgeValue;
    }
}

File: android/src/main/java/com/cactuslab/capacitor/nativenavigation/types/ComponentOptions.kt
package com.cactuslab.capacitor.nativenavigation.types

import com.getcapacitor.JSObject

class ComponentOptions(val title: String?, val stack: StackConfig?, val tab: TabConfig?, val modalPresentationStyle: ModalPresentationStyle?) {
    companion object {
        fun fromJSObject(jsObject: JSObject): ComponentOptions {
            val title = jsObject.getString("title")

            val stack = jsObject.getJSObject("stack")?.let { StackConfig.fromJSObject(it) }
            val tab = jsObject.getJSObject("tab")?.let { TabConfig.fromJSObject(it) }
            val modalPresentationStyle = jsObject.getString("modalPresentationStyle")?.let { ModalPresentationStyle.get(it) }

            return ComponentOptions(title, stack, tab, modalPresentationStyle)
        }
    }
}